src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/BundleMapper.java

```diff
@@ -25,9 +25,15 @@ public Bundle mapToBundle(final PathologyRecord pathologyRecord) {
         Bundle bundle = generateInitialPathologyBundle();
 
         bundle.addEntry()
-            .setFullUrl(FULL_URL_PREFIX.concat(uuidGenerator.generateUUID()))
+            .setFullUrl(FULL_URL_PREFIX.concat(pathologyRecord.getRequester().getId()))
             .setResource(pathologyRecord.getRequester());
 
+        if (pathologyRecord.getPerformer() != null) {
+            bundle.addEntry()
+                .setFullUrl(FULL_URL_PREFIX.concat(pathologyRecord.getPerformer().getId()))
+                .setResource(pathologyRecord.getPerformer());
+        }
```

Comment:
We seem to be preferring to avoid structural branching. Consider:
```suggestion
        Optional.ofNullable(pathologyRecord.getPerformer()).ifPresent(performer ->
            bundle.addEntry()
                .setFullUrl(FULL_URL_PREFIX.concat(performer.getId()))
                .setResource(performer));
        }
```

Reply:
Done in 4bbb811